### DISCUSSION TITLE
New name for "Mixer" class

### DISCUSSION THREAD

**Discussion by JohannesLorenz** [LMMS/lmms#6089] [Project Management]
The reorg proposes to rename `FX-Mixer -> Mixer`, which requires us to find a new name for `Mixer`.

**Which name should we take? Or should we keep the current name?**

Note: This is a discussion, so feel free to add proposals or upvote existing ones.

**Comment by JohannesLorenz** [comment]
Keep the current name `Mixer` for `Mixer` (i.e. `FX-Mixer` can not be named `Mixer`).

**Reply by he29-net** [reply]
If the current `Mixer` handles low-level audio mixing, `Mixer` seems like a good, descriptive name for it, so I would not disregard this option right away.
I assume the incentive for renaming `FX-Mixer` is that it does not deal only with effects. Perhaps as an alternative, it could be renamed to something else that still reflects its higher-level nature and does not limit it to effects. Like `SongMixer`, or `TrackMixer`. Or name one `Mixer` and the other `Blender`, I don't know. Someone with more detailed knowledge about what _exactly_ each of them does could possibly come up with some better suggestions.

**Comment by JohannesLorenz** [comment]
`Audio`

Because it runs the audio threads.

**Reply by PhysSong** [reply]
IMO, `AudioEngine` sounds better.

**Reply by JohannesLorenz** [reply]
Sounds good, though could it sound a bit confusing if we have `Engine` and `AudioEngine`?

**Reply by PhysSong** [reply]
Something like `AudioRenderEngine` then? The key point is that `Audio` itself doesn't describe what it does well.

**Reply by JohannesLorenz** [reply]
That's too long to write. I'd prefer `AudioEngine` then. Do you want to add it as a voting option?

**Reply by PhysSong** [reply]
If you're talking about `AudioEngine`, yes.

**Reply by JohannesLorenz** [reply]
Go ahead :+1:

**Reply by he29-net** [reply]
Or even `AudioMixer`, to keep its current (relatively well matching) descriptive name and make it more specific to distinguish it from the "generic top-level `Mixer`".

**Comment by JohannesLorenz** [comment]
`Realtime`

It contains the realtime threads and the parameters which should be owned by them.

**Comment by JohannesLorenz** [comment]
`AudioEngine`

Proposal from @PhysSong

**Reply by rdrpenguin04** [reply]
IMO, if FX Mixer goes to Mixer, and Mixer goes to AudioMixer, that could get confusing as to which is which. AudioEngine conveys that it's at a lower level.

**Reply by JohannesLorenz** [reply]
`AudioEngine` is the winner. We'll take this.

**Comment by JohannesLorenz** [comment]
`AudioMixer`

Proposal from @he29-net . It keeps its current (relatively well matching) descriptive name and make it more specific to distinguish it from the "generic top-level Mixer".

**Comment by Veratil** [comment]
Just to share the terminology from https://en.wikipedia.org/wiki/Mixing_console

```
A mixing console is also known as an audio mixer,
 audio console, mixing desk, sound mixer, soundboard,
 or simply as board or mixer.
```

**Reply by Spekular** [reply]
I believe these terms would be more applicable for what we currently call the FX mixer, no? When it comes to that I'm very fond of the existing proposal to simply go with "Mixer".

**Reply by Veratil** [reply]
Yeah that's my oops. 😁

**Comment by Spekular** [comment]
`RenderEngine` / `RenderManager`

I haven't delved super deep into the file, but it seems to me that it deals with taking project data (play handles and such) and converting it to samples (/buffers filled with samples). "Rendering" seems appropriate to describe this, and there are references to "rendering" in the file already.

**Reply by rdrpenguin04** [reply]
That is true; the only thing I have against this is that rendering typically refers to video in an AV context, and it's entirely conceivable that a different UI would be using a low-level `RenderEngine`. Plus, if LMMS ever becomes some people's dream as a MultiMedia creation platform, that would be another opportunity for confusion.

I don't hate the idea, but I don't think it's the best name in a forward-thinking way.

**Comment by MaciejMalczyk** [comment]
What about "Mux"? Short and easy name which is connected with signals and signal routing.

**Comment by Rossmaxx** [comment]
Since this is done, I'm closing this discussion out.
